features/events/traps/src/main/java/org/opennms/netmgt/trapd/TrapdConfigBean.java

```diff
@@ -82,4 +89,11 @@ public boolean getNewSuspectOnTrap() {
 		return m_snmpV3Users;
 	}
 
+	@Override
+	public void onUpdate(TrapdConfiguration config) {
+		this.m_snmpTrapAddress = config.getSnmpTrapAddress();
+		this.m_snmpTrapPort =config.getSnmpTrapPort();
+		this.m_snmpV3Users=new ArrayList<SnmpV3User>(TrapReceiverImpl.addToSnmpV3Users(config).values());
```

Comment:
I'm working on this issue. The collection needs to have proper synchronization and I see some other problems in the class.